sa/sa_test.go

```diff
@@ -2574,3 +2578,139 @@ func TestIncidentsForSerial(t *testing.T) {
 	test.AssertNotError(t, err, "Failed to retrieve incidents for serial")
 	test.Assert(t, len(incidentsForSerial) == 1, "No active incidents returned")
 }
+
+type inmemSA struct {
+	sapb.StorageAuthorityClient
+	Impl *SQLStorageAuthority
+}
+
+type sfiData struct {
+	serial *sapb.IncidentSerial
+	err    error
+}
+
+type sfiClient struct {
+	// grpc.ClientStream is embedded to meet the
+	// StorageAuthority_SerialsForIncidentClient interface.
+	grpc.ClientStream
+	sfiChan <-chan sfiData
+}
+
+// Recv receives a single serial from the server.
+func (c sfiClient) Recv() (*sapb.IncidentSerial, error) {
+	sfiData := <-c.sfiChan
+	return sfiData.serial, sfiData.err
+}
+
+type sfiServer struct {
+	// grpc.ServerStream is embedded to meet the
+	// StorageAuthority_SerialsForIncidentServer interface.
+	grpc.ServerStream
+	context context.Context
+	sfiChan chan<- sfiData
+}
+
+// Send sends a single serial to the client.
+func (s sfiServer) Send(serial *sapb.IncidentSerial) error {
+	s.sfiChan <- sfiData{serial, nil}
+	return nil
+}
+
+// Context returns the context originally passed by the client.
+func (s sfiServer) Context() context.Context {
+	return s.context
+}
+
+func (s sfiServer) sendErrAndClose(err error) error {
+	s.sfiChan <- sfiData{nil, err}
+	close(s.sfiChan)
+	return nil
+}
+
+func makeinmemSAGRPCStreamAdapter(ctx context.Context) (sfiClient, sfiServer) {
+	sfiDataChan := make(chan sfiData)
+	return sfiClient{sfiChan: sfiDataChan}, sfiServer{sfiChan: sfiDataChan, context: ctx}
+}
+
+func (s inmemSA) SerialsForIncident(ctx context.Context, req *sapb.SerialsForIncidentRequest, _ ...grpc.CallOption) (sapb.StorageAuthority_SerialsForIncidentClient, error) {
+	client, server := makeinmemSAGRPCStreamAdapter(ctx)
+	go func() {
+		err := s.Impl.SerialsForIncident(req, server)
+		if err != nil {
+			server.sendErrAndClose(err)
+		} else {
+			server.sendErrAndClose(io.EOF)
+		}
+	}()
+	return client, nil
+}
+
+func TestSerialsForIncident(t *testing.T) {
```

Comment:
We should also add a test case for when the table doesn't exist, and when the table name is malformed.